cmd/bench.go

```diff
@@ -281,20 +282,36 @@ func benchE2E(ctx context.Context, args []string, params benchmarkCommandParams,
 		paths = append(paths, params.bundlePaths.v...)
 	}
 
-	rtParams := runtime.Params{
-		Addrs:                  &[]string{fmt.Sprintf("%s:%s", host, port)},
-		Paths:                  paths,
-		Logger:                 logger,
-		BundleMode:             params.bundlePaths.isFlagSet(),
-		SkipBundleVerification: true,
-		EnableVersionCheck:     false,
-		GracefulShutdownPeriod: params.gracefulShutdownPeriod,
-		ShutdownWaitPeriod:     params.shutdownWaitPeriod,
-	}
-
-	rt, err := runtime.NewRuntime(ctx, rtParams)
-	if err != nil {
-		return err
+	// Because of test concurrency, several instances of this function can be
+	// running simultaneously, which will result in occasional collisions when
+	// two goroutines wish to bind the same port for the runtime.
+	// We fix the issue here by simply incrementing the port number if an
+	// "address already in use" error happens.
```

Comment:
@philipaconrad the issue is with how the tests are run and not something users will experience when running the command, correct? If we added a new flag to the command to specify the port to use in the `e2e` mode and default to `18181` and then update the tests to use a random port number, will this resolve the issue?

Reply:
We could update every test, and I think that would work as well. The `benchE2E` function is at the bottom of a fairly shallow call chain, so passing down the `port` parameter wouldn't be too difficult, it just will look out of place relative to the other arguments around it.

Reply:
@ashutosh-narkar I chose to implement @srenatus's suggestion from below, where a random port is bound each time the `benchE2E` function is called.

I think if we want to force a particular port number for the `bench` command, adding a CLI flag like you'd suggested would be the way to go.